packages/next-on-pages/templates/in-memory-mutex.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * What is this file for?
+ *
+ * workerd isolates share the same modules registry meaning that it can happen for the same isolate to run
+ * multiple instances of the same worker and those will share the modules' state.
```

Comment:
An isolate by definition runs only one "instance" of the Worker. However, one Worker instance may handle many requests, possibly at the same time, possibly from different users. Global state will be shared across all these requests.

Basically all servers work this way, including Node.js. We actually would like to change this in Workers, but it requires building new fundamental technology.